src/rational/mat_q/vector/norm.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module includes functionality to compute several norms
+//! defined on vectors.
+
+use super::super::MatQ;
+use crate::{
+    error::MathError,
+    rational::Q,
+    traits::{GetNumColumns, GetNumRows},
+};
+use flint_sys::{
+    fmpq::{fmpq, fmpq_abs, fmpq_addmul, fmpq_cmp},
+    fmpz::fmpz,
+};
+
+impl MatQ {
+    /// Returns the squared Euclidean norm or 2-norm of the given (row or column) vector.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::rational::MatQ;
+    /// use std::str::FromStr;
+    /// # use qfall_math::rational::Q;
```

Comment:
Why hide this?

Reply:
It's not an import that's important for the function, but I changed it.